plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/BulkByScrollTask.java

```diff
@@ -0,0 +1,321 @@
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.lang.Math.min;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.unmodifiableList;
+import static org.elasticsearch.action.search.ShardSearchFailure.readShardSearchFailure;
+
+/**
+ * Task storing information about a currently running BulkByScroll request.
+ */
+public class BulkByScrollTask extends Task {
+    /**
+     * The total number of documents this request will process. 0 means we don't yet know or, possibly, there are actually 0 documents
+     * to process. Its ok that these have the same meaning because any request with 0 actual documents should be quite short lived.
+     */
+    private final AtomicLong total = new AtomicLong(0);
+    private final AtomicLong updated = new AtomicLong(0);
+    private final AtomicLong created = new AtomicLong(0);
+    private final AtomicLong deleted = new AtomicLong(0);
+    private final AtomicLong noops = new AtomicLong(0);
+    private final AtomicInteger batch = new AtomicInteger(0);
+    private final AtomicLong versionConflicts = new AtomicLong(0);
+    private final AtomicReference<List<Failure>> indexingFailures = new AtomicReference<>(emptyList());
+    private final AtomicReference<List<ShardSearchFailure>> searchFailures = new AtomicReference<>(emptyList());
+
+    public BulkByScrollTask(long id, String type, String action, Provider<String> description) {
+        super(id, type, action, description);
+    }
+
+    @Override
+    public Status getStatus() {
+        return new Status(total.get(), updated.get(), created.get(), deleted.get(), batch.get(), versionConflicts.get(), noops.get(),
+                indexingFailures.get(), searchFailures.get());
+    }
+
+    /**
+     * Total number of successfully processed documents.
+     */
+    public long getSuccessfullyProcessed() {
+        return updated.get() + created.get() + deleted.get();
+    }
+
+    /**
+     * All indexing failures.
+     */
+    public List<Failure> getIndexingFailures() {
+        return indexingFailures.get();
+    }
+
+    public static class Status implements Task.Status {
+        public static final Status PROTOTYPE = new Status(0, 0, 0, 0, 0, 0, 0, emptyList(), emptyList());
+
+        private final long total;
+        private final long updated;
+        private final long created;
+        private final long deleted;
+        private final int batches;
+        private final long versionConflicts;
+        private final long noops;
+        private final List<Failure> indexingFailures;
+        private final List<ShardSearchFailure> searchFailures;
+
+        public Status(long total, long updated, long created, long deleted, int batches, long versionConflicts, long noops,
+                List<Failure> indexingFailures, List<ShardSearchFailure> searchFailures) {
+            this.total = total;
```

Comment:
I can certainly throw an exception if any of them aren't >= 0 so the vlongs don't blow up. That seems pretty sane.